components/channel_selector_modal/channel_selector_modal.tsx

```diff
@@ -210,12 +223,12 @@ export default class ChannelSelectorModal extends React.PureComponent {
                     </Modal.Title>
                 </Modal.Header>
                 <Modal.Body>
-                    <MultiSelect
+                    <MultiSelect<ChannelWithTeamData & Value>
```

Comment:
For readability, I would suggest defining the type of this intersection at the beginning of this file so it can be reused in several places here. Maybe something like:

```
type ChannelWithTeamDataValue = ChannelWithTeamData & Value;
```

Reply:
I've updated the PR with this suggestion.